### PR TITLE
[git] Sort the status entries alphabetically, fixed another concurrency issue (fixes #871 (hopefully))

### LINKED ISSUE #871: [git] problem with reported untracked files
It doesn't happen all the time, maybe. But once it starts... Sometimes untracked files are reported twice, sometimes they're not reported at all. 

theia-git-extension--double entries for new files

### PR DESCRIPTION
**Pull request by svenefftinge**
Signed-off-by: Sven Efftinge <sven.efftinge@typefox.io>

### CODE REVIEW

**Comment by marcdumais-work** [comment]
I gave this a quick try to see if it helps with #871, and it seems it doesn't.

**Comment by svenefftinge** [comment]
> I gave this a quick try to see if it helps with #871, and it seems it doesn't.

Could you please provide details on how you reproduce the issue, because I can't.

Anyhow, I think this change should be merged, since it obviously will no longer be possible to have state coming from more than one status event.

**Comment by svenefftinge** [comment]
> I wonder could it happen that the second call of updateView can win over the first one, then the first will be last and will set the outdated state

That could happen with every event that is processed async. I don't think it happens, but worst case, there would be second where outdated state is shown. So not really problematic.

**Comment by marcdumais-work** [comment]
@svenefftinge For me it's extremely easy to reproduce: just start theia on a workspace that has some new files, and it happens right away (I do not see the issue with modified files at this time).

**Comment by svenefftinge** [comment]
> @svenefftinge For me it's extremely easy to reproduce: just start theia on a workspace that has some new files, and it happens right away (I do not see the issue with modified files at this time).

Ok, that helped to reproduce it before this change but not when this change has been applied.
Are you sure you can?

git-widget

**Comment by svenefftinge** [comment]
@akosyakov did you try to reproduce #871 with this change?

**Comment by marcdumais-work** [comment]
re: reproducing #871: it could be that I was just lucky to reproduce easily. Let me try again from a fresh clone and see what happens.

**Comment by marcdumais-work** [comment]
@svenefftinge I can still reproduce easily, from a fresh clone. As I noticed before, it seems that only new files are listed twice - modified files appear only once it seems: 

peek 2017-11-28 06-11

**Comment by svenefftinge** [comment]
@marcdumais-work Are you sure you tried with the PR state and fully rebuilt it? As you can see in my small gif (above), I also tried with adding new files.

**Comment by marcdumais-work** [comment]
@svenefftinge Right - I was on the wrong branch. It does look like your fix has helped. I can't easily reproduce this issue any more, thanks!

**Comment by hexa00** [comment]
@marcdumais-work does this mean you can't reproduce it at all or that it's harder ?

**Comment by marcdumais-work** [comment]
@hexa00 What I mean is that I was not able to reproduce the issue, but I have not extensively tested.
